ui/src/app/applications/components/application-resource-tree/application-resource-tree.tsx

```diff
@@ -116,7 +116,22 @@ function filterGraph(app: models.Application, filteredIndicatorParent: string, g
 }
 
 function compareNodes(first: ResourceTreeNode, second: ResourceTreeNode) {
-    return `${(first.orphaned && '1') || '0'}/${nodeKey(first)}`.localeCompare(`${(second.orphaned && '1') || '0'}/${nodeKey(second)}`);
+    function orphanedToInt(orphaned?: boolean) {
+        return (orphaned && 1) || 0;
+    }
+    function compareRevision(a: string, b: string) {
+        const numberA = Number(a);
+        const numberB = Number(b);
+        if (isNaN(numberA) || isNaN(numberB)) {
+            return a.localeCompare(b);
+        }
+        return Math.sign(numberA - numberB);
+    }
+    return (
+        orphanedToInt(first.orphaned) - orphanedToInt(second.orphaned) ||
+        compareRevision(first.resourceVersion, second.resourceVersion) ||
```

Comment:
I think we want to sort resources of same group/kind by revision independently and latest revision should go first:
 
```
Deployment rev 2
Deployment rev 1
Rollout rev 2
Rollout rev 1
```

instead of 

```
Deployment rev 2
Rollout rev 2
Deployment rev 1
Rollout rev 1
```

In this case we need to switch `compareRevision(first.resourceVersion, second.resourceVersion)` and `nodeKey(first).localeCompare(nodeKey(second))`. 

```
        orphanedToInt(first.orphaned) - orphanedToInt(second.orphaned) ||
        nodeKey(first).localeCompare(nodeKey(second)) ||
        compareRevision(first.resourceVersion, second.resourceVersion)
```

Reply:
@alexmt: I think this is now implemented, but I'm not sure I can construct a proper test for this, can you provide sample data?